src/NServiceBus.Gateway.AcceptanceTests/GatewayEndpoint.cs

```diff
@@ -3,19 +3,24 @@
     using System;
     using System.Threading.Tasks;
     using AcceptanceTesting.Support;
+    using NServiceBus.Configuration.AdvancedExtensibility;
 
     public class GatewayEndpoint : GatewayEndpointWithNoStorage
     {
         public override Task<EndpointConfiguration> GetConfiguration(RunDescriptor runDescriptor, EndpointCustomizationConfiguration endpointCustomizationConfiguration, Action<EndpointConfiguration> configurationBuilderCustomization)
         {
             return base.GetConfiguration(runDescriptor, endpointCustomizationConfiguration, configuration =>
             {
-                GatewayTestSuiteConstraints.Current.ConfigureDeduplicationStorage(
-                    endpointCustomizationConfiguration.CustomEndpointName, 
-                    configuration, 
+                var deduplicationConfiguration = GatewayTestSuiteConstraints.Current.ConfigureDeduplicationStorage(
+                    endpointCustomizationConfiguration.CustomEndpointName,
+                    configuration,
                     runDescriptor.Settings)
                     .GetAwaiter().GetResult();
```

Comment:
This is ugly but I guess we have to allow async in the "configure" stage since we don't have a `runDescriptor.OnTestStarting` hook?